src/trans.h

```diff
@@ -1,26 +1,94 @@
 #ifndef GAP_TRANS_H
 #define GAP_TRANS_H
 
+#include <src/plist.h>
+
 extern UInt INIT_TRANS2(Obj f);
 extern UInt INIT_TRANS4(Obj f);
 
-#define IMG_TRANS(f)      (ADDR_OBJ(f)[0])
-#define KER_TRANS(f)      (ADDR_OBJ(f)[1])
-#define EXT_TRANS(f)      (ADDR_OBJ(f)[2])
-
-#define NEW_TRANS2(deg)   NewBag(T_TRANS2, deg*sizeof(UInt2)+3*sizeof(Obj))
-#define ADDR_TRANS2(f)    ((UInt2*)((Obj*)(ADDR_OBJ(f))+3))
-#define DEG_TRANS2(f)     ((UInt)(SIZE_OBJ(f)-3*sizeof(Obj))/sizeof(UInt2))
-#define RANK_TRANS2(f)    (IMG_TRANS(f)==NULL?INIT_TRANS2(f):LEN_PLIST(IMG_TRANS(f)))
-
-#define NEW_TRANS4(deg)   NewBag(T_TRANS4, deg*sizeof(UInt4)+3*sizeof(Obj))
-#define ADDR_TRANS4(f)    ((UInt4*)((Obj*)(ADDR_OBJ(f))+3))
-#define DEG_TRANS4(f)     ((UInt)(SIZE_OBJ(f)-3*sizeof(Obj))/sizeof(UInt4))
-#define RANK_TRANS4(f)    (IMG_TRANS(f)==NULL?INIT_TRANS4(f):LEN_PLIST(IMG_TRANS(f)))
-
-#define IS_TRANS(f)       (TNUM_OBJ(f)==T_TRANS2||TNUM_OBJ(f)==T_TRANS4)
-#define RANK_TRANS(f)     (TNUM_OBJ(f)==T_TRANS2?RANK_TRANS2(f):RANK_TRANS4(f))
-#define DEG_TRANS(f)      (TNUM_OBJ(f)==T_TRANS2?DEG_TRANS2(f):DEG_TRANS4(f))
+static inline int IS_TRANS(Obj f)
+{
+    return (TNUM_OBJ(f) == T_TRANS2 || TNUM_OBJ(f) == T_TRANS4);
+}
+
+static inline Obj IMG_TRANS(Obj f)
+{
+    GAP_ASSERT(IS_TRANS(f));
+    return ADDR_OBJ(f)[0];
+}
+
+static inline Obj KER_TRANS(Obj f)
+{
+    GAP_ASSERT(IS_TRANS(f));
+    return ADDR_OBJ(f)[1];
+}
+
+static inline Obj EXT_TRANS(Obj f)
+{
+    GAP_ASSERT(IS_TRANS(f));
+    return ADDR_OBJ(f)[2];
+}
+
+static inline Obj NEW_TRANS2(UInt deg)
+{
+    GAP_ASSERT(deg <= 65536);
+    return NewBag(T_TRANS2, deg * sizeof(UInt2) + 3 * sizeof(Obj));
+}
+
+static inline UInt2 * ADDR_TRANS2(Obj f)
+{
+    GAP_ASSERT(TNUM_OBJ(f) == T_TRANS2);
+    return ((UInt2 *)((Obj *)(ADDR_OBJ(f)) + 3));
+}
+
+static inline UInt DEG_TRANS2(Obj f)
+{
+    GAP_ASSERT(TNUM_OBJ(f) == T_TRANS2);
+    return ((UInt)(SIZE_OBJ(f) - 3 * sizeof(Obj)) / sizeof(UInt2));
+}
+
+static inline UInt RANK_TRANS2(Obj f)
+{
+    GAP_ASSERT(TNUM_OBJ(f) == T_TRANS2);
+    return (IMG_TRANS(f) == NULL ? INIT_TRANS2(f) : LEN_PLIST(IMG_TRANS(f)));
+}
+
+static inline Obj NEW_TRANS4(UInt deg)
+{
+    // No assert here since we allow creating new T_TRANS4's when the degree
```

Comment:
Super nitpick, but: shouldn't it be `T_TRANS4s` (plural) instead of `T_TRANS4's` (possessive) ?